src/overlays/actors/ovl_En_Attack_Niw/z_en_attack_niw.c

```diff
@@ -27,23 +36,371 @@ const ActorInit En_Attack_Niw_InitVars = {
+#endif
+
+void func_809B59B0(EnAttackNiw* this, GlobalContext* globalCtx) {
+    if (!func_809B55EC(this, globalCtx)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->unk_25A == 0) {
+            this->unk_25A = 3;
+            this->actor.velocity.y = 3.5f;
+        }
+        if (this->actor.gravity != -2.0f) {
+            this->unk_280 = this->unk_278 = 14000.0f;
+            this->unk_2D0 = this->unk_26C = this->unk_268 = this->unk_284 = this->unk_27C = 0.0f;
+            this->unk_2D4 = this->actor.yawTowardsLink;
+            this->unk_262 = 0x32;
+            this->unk_25C = 0x64;
+            this->actor.gravity = -2.0f;
+        }
+    }
+    if (this->unk_25C == 0x32) {
+        this->unk_2D4 = Math_Rand_CenteredFloat(200.0f) + this->actor.yawTowardsLink;
+    }
+    Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->unk_2D4, 2, this->unk_2DC, 0);
+    Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.x, this->unk_2D0, 2, this->unk_2DC, 0);
+    Math_SmoothScaleMaxF(&this->unk_2DC, 10000.0f, 1.0f, 1000.0f);
+    Math_SmoothScaleMaxF(&this->actor.speedXZ, this->unk_2E0, 0.9f, 1.0f);
+    if ((this->actor.gravity == -2.0f) && (this->unk_262 == 0) &&
+        ((this->actor.bgCheckFlags & 8) || (this->unk_25C == 0))) {
+        this->unk_2E0 = 0.0f;
+        this->actor.gravity = 0.0f;
+        this->unk_2DC = 0.0f;
+        this->unk_2D0 = this->actor.posRot.rot.x - 5000.0f;
+        this->actionFunc = func_809B5C18;
+    } else if (this->actor.bgCheckFlags & 1) {
+        func_809B5268(this, globalCtx, 5);
+    } else {
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+
+void func_809B5C18(EnAttackNiw* this, GlobalContext* globalCtx) {
+    if (!func_809B55EC(this, globalCtx)) {
+        Actor_Kill(&this->actor);
+    } else {
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.x, this->unk_2D0, 5, this->unk_2DC, 0);
+        Math_SmoothScaleMaxF(&this->unk_2DC, 5000.0f, 1.0f, 100.0f);
+        Math_SmoothScaleMaxF(&this->actor.velocity.y, 5.0f, 0.3f, 1.0f);
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+
+void EnAttackNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 tmpf1;
+    EnAttackNiw* this = THIS;
+    EnNiw* attachedActor;
+    Player* player = PLAYER;
+    s32 pad;
+    Vec3f sp30;
+    GlobalContext* ctx2 = globalCtx;
+
+    this->unk_28C++;
+    if (this->unk_254 != 0) {
+        this->unk_254--;
+    }
+    if (this->unk_258 != 0) {
+        this->unk_258--;
+    }
+    if (this->unk_25A != 0) {
+        this->unk_25A--;
+    }
+    if (this->unk_25E != 0) {
+        this->unk_25E--;
+    }
+    if (this->unk_260 != 0) {
+        this->unk_260--;
+    }
+    if (this->unk_25C != 0) {
+        this->unk_25C--;
+    }
+    if (this->unk_262 != 0) {
+        this->unk_262--;
+    }
+
+    this->actor.shape.rot = this->actor.posRot.rot;
+    this->actor.shape.unk_10 = 15.0f;
+    this->actionFunc(this, ctx2);
```

Comment:
id make the temp globalCtx2 (also, ew)
but yea ctx doesnt mean much cause we name alot of things after contexts